consent_manager/api.rst

```diff
@@ -0,0 +1,250 @@
+So if you want use own scripts outside Tag Manager, you need update snippet to use `ppms.cm.api`, or use `dataLayer` interface if replacing snippet is not possible:
+
+.. code-block:: html
+
+    <!DOCTYPE html>
+    <html>
+        <head>
+        <meta charset="UTF-8">
+        <title></title>
+    </head>
+
+    <!-- Start Piwik PRO Tag Manager code -->
+    <script>
+        // Tag Manager async code snippet
+    </script>
+    <!-- End Piwik PRO Tag Manager code -->
+
+    <script>
+        // API call outside Tag Manager injected manually
+        dataLayer.push({event: command, ...args});
+    </script>
+
+    <body>
+        Rest content of document
+    </body>
+
+    </html>
+
+
+Commands
+--------
+All commands work in context of the current visitor and website. Additionally they sometimes require communication with a PPMS server and are asynchronous. Callback functions are used to provide response value or information about errors. `onSuccess(...args)` callback is always required. `onFailure(exception)` callback is optional and if is specified, any error object occurred will be passed as a argument. If not specified, error is reported directly on console output.
+
+Get consent types
+`````````````````````````````````````
+Fetches a list of consent types.
+
+Code::
+
+    ppms.cm.api('getComplianceTypes', onFulfilled, onRejected);
+    dataLayer.push({'event': 'ppms.cm:getComplianceTypes', parameters: [onFulfilled, onRejected]});
+
+.. function:: onFulfilled(types)
+
+    The fulfilment handler callback (called with result). This function is **required**.
+
+    :param Array<string> types: **Required** Array of consent types
+
+        Example::
+
+            ["remarketing", "analytics"]
+
+.. function:: onRejected(error)
+
+    The rejection handler callback (called with error code). If not specified, exception will be thrown in main stacktrace.
+
+    :param string|object error: **Required** Error code or exception.
+
+Get new consent types
+`````````````````````````````````````
+Fetches a list of new consent types which were appearing after given consents.
+
+Code::
+
+    ppms.cm.api('getNewComplianceTypes', onFulfilled, onRejected);
+    dataLayer.push({'event': 'ppms.cm:getNewComplianceTypes', parameters: [onFulfilled, onRejected]});
+
+.. function:: onFulfilled(types)
+
+    The fulfilment handler callback (called with result).
+
+    :param Array<string> types: **Required** Array of consent types
+
+        Example::
+
+            ["remarketing", "analytics"]
+
+.. function:: onRejected(error)
+
+    The rejection handler callback (called with error code).
+
+    :param string|object error: **Required** Error code or exception.
+
+
+Set compliance settings
+`````````````````````````````````````
+Set compliance settings base on user decision.
+This API command might be useful when user interact with custom, extended UI that react on user approve/reject action.
```

Comment:
`extended UI that react on user` => `extended UI that reacts on user`?
Language here is questionable.